lib/Auth/Source/Radius.php

```diff
@@ -133,77 +140,47 @@ public function __construct(array $info, array $config)
      */
     protected function login(string $username, string $password): array
     {
-        $radius = radius_auth_open();
-        if (!is_resource($radius)) {
-            throw new Exception("Insufficient memory available to create handle.");
-        }
+        $radius = new RadiusClient();
+        $response = false;
 
         // Try to add all radius servers, trigger a failure if no one works
-        $success = false;
         foreach ($this->servers as $server) {
-            if (!isset($server['port'])) {
-                $server['port'] = 1812;
-            }
-            if (
-                !radius_add_server(
-                    $radius,
-                    $server['hostname'],
-                    $server['port'],
-                    $server['secret'],
-                    $this->timeout,
-                    $this->retries
-                )
-            ) {
-                Logger::info(
-                    "Could not add radius server: " . radius_strerror($radius)
-                );
-                continue;
-            }
-            $success = true;
-        }
-        if (!$success) {
-            throw new Exception('Error adding radius servers, no servers available');
-        }
+            $radius->setServer($server['hostname']);
+            $radius->setAuthenticationPort($server['port']);
+            $radius->setSecret($server['secret']);
+            $radius->setDebug($this->debug);
+            $radius->setTimeout($this->timeout);
 
-        if (!radius_create_request($radius, \RADIUS_ACCESS_REQUEST)) {
-            throw new Exception(
-                'Error creating radius request: ' . radius_strerror($radius)
-            );
-        }
+            $httpUtils = new Utils\HTTP();
+            $radius->setNasIpAddress($httpUtils->getSelfHost());
```

Comment:
This fails in my (develop) situation because SelfHost is not resolvable on the server.

This makes it more robust and does not require the name to be looked up on every auth.
```suggestion
            $radius->setNasIpAddress($_['SERVER_ADDR'] ?: $httpUtils->getSelfHost());
```

(We might make a httpUtil for getting the SERVER_ADDR)

I might send a PR to the radius lib to prefer SERVER_ADDR over lookups if available.

Reply:
Changed this now in the straightforward way